src/Transport/Topology/MetaModel/EntityInfo.cs

```diff
@@ -23,17 +22,12 @@ public EntityInfo()
 
         protected bool Equals(EntityInfo other)
         {
-            if (Type == EntityType.Subscription)
-            {
-                var entity = RelationShips.First(r => r.Type == EntityRelationShipType.Subscription);
-                var otherEntity = other.RelationShips.First(r => r.Type == EntityRelationShipType.Subscription);
-                var targetPathEquals = string.Equals(entity.Target.Path, otherEntity.Target.Path);
-                if (!targetPathEquals)
-                {
-                    return false;
-                }
-            }
-            return string.Equals(Path, other.Path) && Type == other.Type && Equals(Namespace, other.Namespace);
+            return string.Equals(Path, other.Path) && Type == other.Type && Equals(Namespace, other.Namespace) && DerivedEqual(other);
```

Comment:
Yes it would. Reason I didn't do that is because we'd end up with two new method signatures  detected by API comparison - `Equals(object)` and `GetHashCode()` required when overriding equality method. That means we're breaking binary compatibility, isn't it @danielmarbach?